File: java/bob/src/main/java/Bob.java
class Bob {
  String hey(String input) {
    String trimmed = input.trim();
    System.out.println(trimmed);
    if (trimmed.length() == 0) {
      return "Fine. Be that way!";
    }
    if (trimmed.matches(".*[a-zA-Z]+.*") && trimmed.toUpperCase().equals(trimmed)) {
      if (trimmed.endsWith("?")) {
        return "Calm down, I know what I'm doing!";
      }
      return "Whoa, chill out!";
    }
    if (trimmed.endsWith("?")) {
      return "Sure.";
    }
    return "Whatever.";
  }
}

File: dart/bob/lib/bob.dart
class Bob {
  String response(String input) {
    input = input.trim();
    if (input == '') {
      return 'Fine. Be that way!';
    }
    if (input.endsWith('?')) {
      if (isYelling(input)) {
        return 'Calm down, I know what I\'m doing!';
      }
      return 'Sure.';
    }
    if (isYelling(input)) {
      return 'Whoa, chill out!';
    }
    return 'Whatever.';
  }
}

bool isYelling(String s) {
  return s.toUpperCase() == s && s.contains(new RegExp(r'[A-Z]'));
}